modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/AccountMenuContent.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayDropDown from '@clayui/drop-down';
+import React, {useContext} from 'react';
+
+import ChartContext from '../ChartContext';
+import {deleteAccount} from '../data/accounts';
+import {PERMISSION_CHECK_ON_HEADLESS_API_ACTIONS} from '../utils/flags';
+
+export default function AccountMenuContent({closeMenu, data, parentData}) {
+	const {chartInstanceRef} = useContext(ChartContext);
+
+	function handleDelete() {
+		if (
+			confirm(
+				Liferay.Util.sub(
+					Liferay.Language.get('x-will-be-deleted'),
+					data.name
+				)
+			)
+		) {
+			deleteAccount(data.id).then(() => {
+				chartInstanceRef.current.deleteNodes([data], true);
+
+				if (parentData) {
+					chartInstanceRef.current.updateNodeContent({
+						...parentData,
+						numberOfAccounts: parentData.numberOfAccounts - 1,
+					});
+				}
+
+				closeMenu();
+			});
+		}
+	}
+
+	const actions = [];
+
+	if (!PERMISSION_CHECK_ON_HEADLESS_API_ACTIONS) {
```

Comment:
Here you can just earlier return an array using 

```jsx
return [(
			<ClayDropDown.Item key="delete" onClick={handleDelete}>
				{Liferay.Language.get('delete')}
			</ClayDropDown.Item>
)]
```

and outside the conditional, you can just return `[]`.

Why it requires an array since it can have one element or nothing?

Reply:
Hi @diegonvs in the next iteration of this PR I've already removed the flag `PERMISSION_CHECK_ON_HEADLESS_API_ACTIONS` so the code is already different.

Reply:
Ok! :)